src/lax.ts

```diff
@@ -0,0 +1,700 @@
+    }
+
+    const driverValues = {}
+
+    this.drivers.forEach((driver) => {
+      driverValues[driver.name] = driver.getValue(this.frame)
+    })
+
+    this.elements.forEach((element) => {
+      element.update(driverValues, this.frame)
+    })
+
+    if (this.debug) {
+      this.debugData.frameLengths.push(Date.now() - this.debugData["frameStart"])
+    }
+
+    if (this.frame % 60 === 0 && this.debug) {
+      const averageFrameTime = Math.ceil((this.debugData.frameLengths.reduce((a, b) => a + b, 0) / 60))
+      console.log(`Average frame calculation time: ${averageFrameTime}ms`)
+      this.debugData.frameLengths = []
+    }
+
+    this.frame++
+
+    window.requestAnimationFrame(this.onAnimationFrame)
+  }
+
+  addDriver = (name: string, getValueFn: () => number, options: DriverOptions = {}) => {
+    this.drivers.push(
+      new LaxDriver(name, getValueFn, options)
+    )
+  }
+
+  removeDriver = (name: string) => {
+    this.drivers = this.drivers.filter(driver => driver.name !== name)
+  }
+
+  findAndAddElements = () => {
+    this.elements = []
+    const elements = document.querySelectorAll(".lax")
+
+    elements.forEach((domElement: HTMLElement | Element) => {
+      const driverName = "scrollY"
+      const presets = []
+
+      domElement.classList.forEach((className) => {
+        if (className.includes("lax_preset")) {
+          const preset = className.replace("lax_preset_", "")
+          presets.push(preset)
+        }
+      })
+
+      const transforms = {
+        [driverName]: {
+          presets
+        }
+      }
+
+      this.elements.push(new LaxElement('.lax', this, domElement, transforms, 0))
+    })
+  }
+
+  addElements = (selector: string, transforms: ElementTransforms, options?: ElementOptions) => {
+    const domElements = document.querySelectorAll(selector)
+
+    domElements.forEach((domElement, i) => {
+      this.elements.push(new LaxElement(selector, this, domElement, transforms, i, options))
+    })
+  }
+
+  removeElements = (selector: string) => {
+    this.elements = this.elements.filter(element => element.selector !== selector)
+  }
+
+  addElement = (domElement: HTMLElement | Element, transforms: ElementTransforms, options?: ElementOptions) => {
+    this.elements.push(new LaxElement('', this, domElement, transforms, 0, options))
+  }
+
+  removeElement = (domElement: HTMLElement | Element) => {
+    this.elements = this.elements.filter(element => element.domElement !== domElement)
+  }
+}
+
+export const laxInstance = new Lax();
+const lax = {
+  addElements: laxInstance.addElements,
+  removeElements: laxInstance.removeElements,
+  removeElement: laxInstance.removeElement,
+  addElement: laxInstance.addElement,
+  init: laxInstance.init,
+  addDriver: laxInstance.addDriver,
+  removeDriver: laxInstance.removeDriver,
+};
+(() => {
+  
+
+  if (typeof module !== 'undefined' && typeof module.exports !== 'undefined')
+    module.exports = lax;
+  else
+    window["lax"] = lax;
+  
```

Comment:
Remove extra spacing

Reply:
I can but remember the typescript compiler takes care of that ;)